### ISSUE TITLE
can   debounce   provide a  config like  lodash's  debounce   能否给debounce提供一个配置项 跟lodash的debounce一样

### ISSUE THREAD

**Issue by pangao66** [vueuse/vueuse#752]
debounce 能否提供一个跟lodash一样的配置项   
lodash里面的debounce 可以配置 触发开始是不是执行一次 这个非常重要,  比如点击按钮 需要做防抖功能, 但是想点击的第一次就触发一次,后面重复点击再防抖, 目前这个useDebounce没法满足 

sometimes  I want click a buttton use debounce ,  but I wan't emit the event at I click the first time, the debounce of debounce  has  provide the leading and the trailing  config

**Comment by harmyderoman** [comment]
You can take `useDebounce` for `{ leading: false, trailing: true }`
[useThrottle](https://vueuse.org/shared/useThrottle/) for `{ leading: true, trailing: true }`
[useThrottleFn ](https://vueuse.org/shared/useThrottleFn/)for { leading: true, trailing: false } if you set it as `useThrottleFn(fn, ms , false)` 3rd param is trailing
I think `useThrottle` also need trailing param to be more useful

**Comment by harmyderoman** [comment]
@patak-js What do you think? Do we need `trailing` param in the `useThrottle `  function?

**Comment by patak-js** [comment]
Yes, maybe it was added to useThrottleFn and didn't get implemented in useThrottle 👍🏼

**Comment by harmyderoman** [comment]
@patak-js Changes can affect many functions

**Comment by patak-js** [comment]
I don't know if I understand you here. But it may be easier to discuss these improvements in draft PRs. IIUC the idea in this case was to add a thrid param to useThrottle but if you see other changes needed, I advice you to send separate PRs so they are easier to merge

**Comment by harmyderoman** [comment]
Ok I'll start working on this

**Comment by harmyderoman** [comment]
PR created [https://github.com/vueuse/vueuse/pull/760](https://github.com/vueuse/vueuse/pull/760)

**Comment by 642315342** [comment]
Why is it only added to useThrottleFn but not to useDebounceFn
